src/Compilers/VisualBasic/Portable/CodeGen/CodeGenerator.vb

```diff
@@ -261,7 +287,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeGen
                 instructionsEmitted = EmitStatementAndCountInstructions(statement)
             End If
 
-            If instructionsEmitted = 0 AndAlso span <> Nothing AndAlso _optimizations = OptimizationLevel.Debug Then
+            If instructionsEmitted = 0 AndAlso span <> Nothing AndAlso _ilEmitStyle = OptimizationLevel.Debug Then
```

Comment:
I will check. The idea is that if no instructions were emitted for the statement (like it was a label), only then we need a nop, since the sequence point needs some code..